### PR TITLE
ST_DumpAsPolygons

### PR DESCRIPTION
**Pull request by ebocher**
This PR adds : 

- ST_DumpAsPolygons with the same signature than Postgis. Thanks to JAITOOLS. In the future, a new vectorize function will be added to support large dataset. The function is ready to extract vector data but doesn't work correctly with H2 due to a problem with SimpleRowSource. 
-  GeoRasterUtils, some useful function to rescale a GeoRaster to byte with a Min and a Max

This PR is pending an evaluation from @nicolas-f  ( in few weeks).

### CODE REVIEW

File: h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     * 
+     * @param geoRaster the input georaster
+     * @param bandIndice the band indice
+     * @param excludeNodata true to exclude nodata value
+     * @return 
+     * @throws java.io.IOException 
+     * @throws java.sql.SQLException 
+     */
+    public static ResultSet vectorizePolygons(GeoRaster geoRaster, int bandIndice, boolean excludeNodata) throws IOException, SQLException{        
+        if(geoRaster==null){
```
Comment (nicolas-f) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java]: You have to check the connection type like here:
https://github.com/orbisgis/h2gis/blob/master/h2network/src/main/java/org/h2gis/network/graph_creator/ST_Accessibility.java#L111

File: h2spatial-ext/src/test/java/org/h2gis/h2spatialext/function/spatial/raster/RasterFunctionTest.java [hunk 2]
```diff
@@ -1396,4 +1397,40 @@ public void testST_FillSinksChezine() throws Exception {
...
+        st.execute("drop table if exists test, vectorize;");
+        st.execute("create table test(id identity, the_raster raster)");
+        // Create table with test image
+        PreparedStatement ps = connection.prepareStatement("INSERT INTO TEST(the_raster) " + "values(?)");
+        ps.setBinaryStream(1, GeoRasterRenderedImage.create(image, pixelSize, -pixelSize, 0, height, 0, 0, 27572, noData)
+                .asWKBRaster());
+        ps.execute();
+        ps.close();
+
+        st.execute("create table vectorize as select ST_DumpAsPolygons(the_raster) FROM test");
```
Comment (nicolas-f) [inline h2spatial-ext/src/test/java/org/h2gis/h2spatialext/function/spatial/raster/RasterFunctionTest.java]: DumpAsPolygons is a table function.

This **may** work following your signature.
```sql
create table vectorize as select * from ST_DumpAsPolygons((SELECT the_raster from test limit 1));
```

File: h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java (hunk 1 above)
Comment (ebocher) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java]: Don't understand this hack

Comment (nicolas-f) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java]: This is not a hack. H2 will call multiple times your function (without calling next) in order to fetch result set metadata.

**Comment by nicolas-f** [comment]
Lol, I don't know how but you have successfully crashed java engine.

File: h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java (hunk 1 above)
Comment (ebocher) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/raster/ST_DumpAsPolygons.java]: Java doc on isColumnListConnection saids "This is a hack. See: https://groups.google.com/forum/#!topic/h2-database/NHH0rDeU258". ;-)
By the way we must add this test for all Resulset function  : GridRowSet, ExplodeResultSet, ST_TriangleContouring,  Right ?

**Comment by ebocher** [comment]
;-)

**Pull request merged**